src/pages/dar_application/DataAccessRequest.js

```diff
@@ -50,12 +50,13 @@ const searchDatasets = (query, callback, currentDatasets) => {
 
 const formatSearchDataset = (ds) => {
   return {
-    key: ds.dataSetId.toString(),
-    value: ds.dataSetId,
+    key: ds.id || ds.datasetId,
+    value: ds.id || ds.datasetId,
     dataset: ds,
-    displayText: ds.datasetIdentifier,
-    label: span({}, [
-      span({style: {fontWeight: 'bold'}}, [ds.datasetIdentifier]), ' | ', ds.name]),
+    displayText: ds.identifier || ds.datasetIdentifier,
+    label: <span>
+      <span style={{fontWeight: 'bold'}}>{ds.identifier || ds.datasetIdentifier}</span> | {ds.name || ds.datasetName}
+    </span>
```

Comment:
The fact that we have `.id`, `.datasetId`, `.identifier`, `.datasetIdentifier`, and also `.dataSetId` is confusing. I think the key and value needs to be `ds.dataSetId`.

Reply:
They are different things depending on the object. I agree, this is confusing, but does reflect reality. To make them all the same, we need to modify the API to make that happen, which is also confusing. Another possibility might be to try and pre-process the objects and conform them to a common standard.

Reply:
Mapping the autocomplete response so they look like the old version of a dataset should simplify these conditions.

Reply:
Looking into the broken component test.